### PR TITLE
MODINVOICE-46 voucherNumber Generation

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODINVOICE-46](https://issues.folio.org/browse/MODINVOICE-46) voucherNumbers should be generated based on the configured prefix and sequence

## Approach
- Refactored the getting configuration logic as follows 
  - replaced `ConfigurationsClient.getConfigurationsEntries()` by `HelperUtils.handleGetRequest()` to be able reuse and close `httpClient`
  - load configuration from the beginning and store in `AbstractHelper.tenantConfiguration` variable
  - get the required configuration from this variable when needed
- updating voucherNumber generation logic to concatenate sequence number with extracted from mod-configuration voucher number prefix if present and valid
- returning error response with appropriate error code

Additionally added to enums missing invoice and invoiceLine protected fields

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java [hunk 1]
```diff
@@ -282,10 +293,32 @@ private Voucher withRequiredFields(Voucher voucher, Invoice invoice) {
     Voucher voucher = new Voucher();
     voucher.setInvoiceId(invoice.getId());
 
-    return voucherHelper.generateVoucherNumber()
+    return getVoucherNumberWithPrefix()
       .thenApply(voucher::withVoucherNumber);
   }
 
+  private CompletableFuture<String> getVoucherNumberWithPrefix() {
+    return voucherHelper.generateVoucherNumber()
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/InvoiceHelper.java]: According to jira description voucher number prefix validation should be done prior to sequence generation

File: src/main/java/org/folio/rest/impl/VoucherLineHelper.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import io.vertx.core.Context;
 import io.vertx.core.json.JsonObject;
 import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/VoucherLineHelper.java]: Unused import

Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/VoucherLineHelper.java]: Removed

File: src/main/java/org/folio/rest/impl/InvoiceHelper.java (hunk 1 above)
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/InvoiceHelper.java]: Done

**Pull request merged**